src/core/mac/sub_mac.cpp

```diff
@@ -1175,9 +1222,9 @@ void SubMac::HandleCslTimer(void)
         else if (mState == kStateCslSample)
         {
             IgnoreError(Get<Radio>().Receive(mCslChannel));
-            LogDebg("CSL sample %lu, duration %lu", ToUlong(mCslTimer.GetNow().GetValue()),
-                    ToUlong(timeAhead + timeAfter));
         }
+
+        LogDebg("CSL sample %lu, duration %lu", ToUlong(mCslTimer.GetNow().GetValue()), ToUlong(timeAhead + timeAfter));
```

Comment:
Well, it depends on the interpretation of "CSL sample", but there is actually an issue here, since the sample time for `ReceiveAt()` is actually `mCslSampleTime.GetValue() - periodUs`. Still the log is valuable to see the duration of the receive window.

Would it be acceptable if I just subtract `periodUs` from `mCslSampleTime` for this log when `RadioSupportsReceiveTiming`? The log would be actually printed some time before the actual sample time, but the timestamp will be correct for the current cycle.

Reply:
I think the most useful log message here (and also original intended use) was to show the time of the start of the CSL-sampling window, and not the "expected CSL sample time" (i.e. expected ideal start of MAC header).  The "expected CSL sample time" is already printed in the `LogReceived` function when a frame is received.

We could call it "CSL sample" in both cases, or alternatively "CSL start" or "CSL window" to avoid confusion.

So the value would be
*  `mCslTimer.GetNow().GetValue()` in case of non-scheduled radio;
*  `mCslSampleTime.GetValue() - periodUs - timeAhead` in case of scheduled radio.  And the message would be printed ahead in time, that's ok.

Reply:
Please re-check the changes in `SubMac::HandleCslTimer`.